staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -95,7 +96,10 @@ var defaultConfigFlags = genericclioptions.NewConfigFlags(true).WithDeprecatedPa
 
 // NewDefaultKubectlCommand creates the `kubectl` command with default arguments
 func NewDefaultKubectlCommand() *cobra.Command {
+	// TODO: add something to check for --kuberc flag and override filepath
+	homeDir, _ := os.UserHomeDir()
```

Comment:
As you said in the TODO already, this should be the same with how `kubeconfig` handles this.